disttask/loaddata/scheduler.go

```diff
@@ -113,40 +121,44 @@ func (s *ImportScheduler) OnSubtaskFinished(ctx context.Context, subtaskMetaByte
 	logutil.BgLogger().Info("OnSubtaskFinished", zap.Any("taskMeta", s.taskMeta))
 	var subtaskMeta SubtaskMeta
 	if err := json.Unmarshal(subtaskMetaBytes, &subtaskMeta); err != nil {
-		return subtaskMetaBytes, err
+		return nil, err
 	}
 
-	dataEngine, ok := s.dataEngines.Load(subtaskMeta.ID)
+	val, ok := s.sharedVars.Load(subtaskMeta.ID)
 	if !ok {
-		return subtaskMetaBytes, errors.Errorf("data engine %d not found", subtaskMeta.ID)
+		return nil, errors.Errorf("sharedVars %d not found", subtaskMeta.ID)
 	}
-	engine, ok := dataEngine.(*backend.OpenedEngine)
+	sharedVars, ok := val.(*SharedVars)
 	if !ok {
-		return subtaskMetaBytes, errors.Errorf("data engine %d not found", subtaskMeta.ID)
+		return nil, errors.Errorf("sharedVars %d not found", subtaskMeta.ID)
 	}
-	closedEngine, err := engine.Close(ctx)
-	if err != nil {
-		return subtaskMetaBytes, err
+
+	logutil.BgLogger().Info("import data engine", zap.Any("id", subtaskMeta.ID))
+	if closedEngine, err := sharedVars.DataEngine.Close(ctx); err != nil {
```

Comment:
we should `close and cleanup engine` in all case, since there's no checkpoint, add a todo?